Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs

```diff
@@ -201,7 +201,7 @@ private void Execute()
                     {
                         try
                         {
-                            ExecuteReplicationOnce(runningBecauseOfDataModifications);
+                            ExecuteReplicationOnce(runningBecauseOfDataModifications).Wait(TimeSpan.FromSeconds(30));
```

Comment:
Why are you waiting for 30 seconds here?
What happens if this takes more than 30 seconds to complete?